src/app.module.ts

```diff
@@ -1,17 +1,27 @@
 import { Module } from '@nestjs/common';
-import { AppController } from './app.controller';
-import { WeatherModule } from './weather/weather.module';
 import { DatabaseModule } from './database/database.module';
 import { ConfigModule } from '@nestjs/config';
+import { APP_GUARD } from '@nestjs/core';
+import { AtGuard } from './common/guards';
+import { AuthModule } from './auth/auth.module';
+import { UsersModule } from './users/users.module';
+import { MailModule } from './mail/mail.module';
 
 @Module({
   imports: [
     DatabaseModule,
     ConfigModule.forRoot({
       isGlobal: true,
     }),
-    WeatherModule,
+    AuthModule,
+    UsersModule,
+    MailModule,
+  ],
+  providers: [
+    {
+      provide: APP_GUARD,
+      useClass: AtGuard,
```

Comment:
I assume `AtGuard` stands for `AccessTokenGuard`. I would use full names, but I would name it differently, like `UserAuthGuard` or `BearerAuthGuard` or something similar.